instrumentation/opentelemetry-instrumentation-mysqlclient/tests/test_mysqlclient_integration.py

```diff
@@ -96,6 +96,264 @@ def test_instrument_connection(self, mock_connect):
         spans_list = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(spans_list), 1)
 
+    @mock.patch("opentelemetry.instrumentation.dbapi.instrument_connection")
+    @mock.patch("MySQLdb.connect")
+    # pylint: disable=unused-argument
+    def test_instrument_connection_enable_commenter_dbapi_kwargs(
+        self,
+        mock_connect,
+        mock_instrument_connection,
+    ):
+        cnx = MySQLdb.connect(database="test")
+        cnx = MySQLClientInstrumentor().instrument_connection(
+            cnx,
+            enable_commenter=True,
+            commenter_options={"foo": True},
+        )
+        cursor = cnx.cursor()
+        cursor.execute("Select 1;")
+        kwargs = mock_instrument_connection.call_args[1]
+        self.assertEqual(kwargs["enable_commenter"], True)
+        self.assertEqual(kwargs["commenter_options"], {"foo": True})
+
+    def test_instrument_connection_with_dbapi_sqlcomment_enabled(self):
+        mock_cursor = mock.MagicMock()
+        mock_cursor.execute = mock.MagicMock()
+        mock_connection = mock.MagicMock()
+        mock_connection.cursor.return_value = mock_cursor
+
+        mock_connect_module = mock.MagicMock()
+        mock_connect_module.__name__ = "MySQLdb"
+        mock_connect_module.threadsafety = "123"
+        mock_connect_module.apilevel = "123"
+        mock_connect_module.paramstyle = "test"
+        mock_connect_module._mysql.get_client_info = mock.Mock(
+            return_value="foobaz"
+        )
+        mock_connect_module.connect = mock.Mock(return_value=mock_connection)
```

Comment:
I think you can remove most of this mock setup boilerplate by relying on Mocks being lazily created and tracked implicitly:

```py
        mock_connect_module = mock.MagicMock(
            __name__="MySQLdb",
            threadsafety="123",
            apilevel="123",
            paramstyle="test",
        )
        mock_connect_module._mysql.get_client_info.return_value = "foobaz"
        mock_cursor = mock_connect_module.connect().cursor()
```

IMO this is more readable. Should work for your other test cases too.

Reply:
Yes!! Changed in [e0ecef0](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2941/commits/e0ecef01b9babd5e9e6ae6c07d09c92360b33e5d)